UI/HtmlControls/HtmlControls/HtmlSelect.cs

```diff
@@ -0,0 +1,118 @@
+            this.ExecuteJavascript("element.addEventListener(\"loaded\", ()=>element.dispatchEvent(new CustomEvent(\"value\", {detail: \"\"+element.selectedIndex})));");
+            this.RegisterHtmlCustomEventHandler("value", OnHtmlSelectionChanged);
+        }
+
+        private void OnHtmlSelectionChanged(object sender, HtmlCustomEventArgs e)
+        {
+            if(int.TryParse(e.Detail, System.Globalization.NumberStyles.Integer | NumberStyles.AllowLeadingSign, NumberFormatInfo.InvariantInfo, out var index))
+            {
+                SelectedIndex = index;
+
+                if (index < 0)
+                {
+                    // No selection
+                    SelectedValue = null;
+                }
+                else
+                {
+                    SelectedValue = Options[index].Value;
+                }
+            }
+        }
+
+        public static readonly DependencyProperty SelectedValueProperty = DependencyProperty.Register(
+            "SelectedValue", typeof(string), typeof(HtmlSelect), new PropertyMetadata(default(string), propertyChangedCallback: OnSelectedValueChanged));
+
+        public string SelectedValue
+        {
+            get => (string)GetValue(SelectedValueProperty);
+            set => SetValue(SelectedValueProperty, value);
+        }
+
+        private static void OnSelectedValueChanged(DependencyObject o, DependencyPropertyChangedEventArgs args)
+        {
+            if (o is HtmlSelect select && args.NewValue is string selectedValue)
+            {
+                // TODO
+            }
+        }
+
+        public static readonly DependencyProperty SelectedIndexProperty = DependencyProperty.Register(
+            "SelectedIndex", typeof(int), typeof(HtmlSelect), new PropertyMetadata(default(int), propertyChangedCallback: OnSelectedIndexChanged));
+
+        public int SelectedIndex
+        {
+            get => (int)GetValue(SelectedIndexProperty);
+            set => SetValue(SelectedIndexProperty, value);
+        }
+
+        private static void OnSelectedIndexChanged(DependencyObject o, DependencyPropertyChangedEventArgs args)
+        {
+            if (o is HtmlSelect select && args.NewValue is int selectedIndex)
+            {
+                select.ExecuteJavascript($"element.selectedIndex={selectedIndex};");
+            }
+        }
+
+        private void OnOptionsChanged(object sender, NotifyCollectionChangedEventArgs e)
+        {
+            // TODO: Need a fix for calling AddChild/RemoveChild
+            //switch(e.Action)
+            //{
+            //    case NotifyCollectionChangedAction.Add:
+            //        {
+            //            var index = e.NewStartingIndex;
+            //            foreach(var newItem in e.NewItems)
+            //            {
+            //                AddChild(newItem as UIElement, index++);
```

Comment:
@carldebilly can you suggest what we should do here as AddChild and RemoveChild aren't accessible as they're marked internal